src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -215,6 +212,9 @@ public override Task AnalyzeProjectAsync(Project project, bool semanticsChanged,
                     projectInfo = await SymbolTreeInfo.GetInfoForSourceAssemblyAsync(
                         project, checksum, cancellationToken).ConfigureAwait(false);
 
+                    Contract.ThrowIfNull(projectInfo);
+                    Contract.ThrowIfTrue(projectInfo.Checksum != checksum, "If we computed a SymbolTreeInfo, then its checksum much match our checksum.");
```

Comment:
:bulb: Use `projectInfo?.Checksum != checksum` to combine these checks.

:bulb: Use this `Contract.Assume` if this condition cannot realistically fail at runtime.

:bulb: Prefer `nameof(SymbolTreeInfo)` in the string.